pathd/path_pcep_controller.c

```diff
@@ -696,17 +809,51 @@ int pcep_thread_event_update_pce_options(struct ctrl_state *ctrl_state,
 			 "failed to update PCC configuration");
 	}
 
+	calculate_best_pce(ctrl_state);
 	return 0;
 }
 
-int pcep_thread_event_remove_pcc(struct ctrl_state *ctrl_state, int pcc_id)
+int pcep_thread_event_remove_pcc_by_id(struct ctrl_state *ctrl_state,
+				       int pcc_id)
 {
-	struct pcc_state *pcc_state;
+	if (pcc_id) {
+		struct pcc_state *pcc_state = get_pcc_state(ctrl_state, pcc_id);
+		if (pcc_state) {
+			remove_pcc_state(ctrl_state, pcc_state);
+			pcep_pcc_finalize(ctrl_state, pcc_state);
+		}
+	}
+	return 0;
+}
+int pcep_thread_event_remove_pcc_all(struct ctrl_state *ctrl_state)
+{
+	if (!ctrl_state) {
+		return 0;
+	}
 
-	if (pcc_id <= ctrl_state->pcc_count) {
-		pcc_state = get_pcc_state(ctrl_state, pcc_id);
-		remove_pcc_state(ctrl_state, pcc_state);
-		pcep_pcc_finalize(ctrl_state, pcc_state);
+	for (int i = 0; i < MAX_PCC; i++) {
+		pcep_thread_event_remove_pcc_by_id(
+			ctrl_state, get_pcc_id_by_idx(ctrl_state, i));
+	}
+	return 0;
+}
+int pcep_thread_event_remove_pcc(struct ctrl_state *ctrl_state,
+				 struct pce_opts *pce_opts)
+{
+	if (!ctrl_state) {
```

Comment:
Same here, could this function be called with ctrl_state being NULL ?

Reply:
done